homeassistant/components/onewire/sensor.py

```diff
@@ -30,33 +39,137 @@
+# via owserver (network protocol)
+
+HOBBYBOARD_EF = {
+    "HobbyBoards_EF": {
+        "humidity": "humidity/humidity_corrected",
+        "humidity_raw": "humidity/humidity_raw",
+        "temperature": "humidity/temperature",
+    },
+    "HB_MOISTURE_METER": {
+        "moisture_0": "moisture/sensor.0",
+        "moisture_1": "moisture/sensor.1",
+        "moisture_2": "moisture/sensor.2",
+        "moisture_3": "moisture/sensor.3",
+    },
+    "HB_HUB": {
+        "branch_0": "hub/branch.0",
+        "branch_1": "hub/branch.1",
+        "branch_2": "hub/branch.2",
+        "branch_3": "hub/branch.3",
+        "short_0": "hub/short.0",
+        "short_1": "hub/short.1",
+        "short_2": "hub/short.2",
+        "short_3": "hub/short.3",
+    },
 }
 
 SENSOR_TYPES = {
     "temperature": ["temperature", TEMP_CELSIUS],
     "humidity": ["humidity", "%"],
+    "humidity_raw": ["humidity", "%"],
     "pressure": ["pressure", "mb"],
     "illuminance": ["illuminance", "lux"],
+    "wetness_0": ["wetness", "%"],
+    "wetness_1": ["wetness", "%"],
+    "wetness_2": ["wetness", "%"],
+    "wetness_3": ["wetness", "%"],
+    "moisture_0": ["moisture", "cb"],
+    "moisture_1": ["moisture", "cb"],
+    "moisture_2": ["moisture", "cb"],
+    "moisture_3": ["moisture", "cb"],
+    "counter_a": ["counter", "count"],
+    "counter_b": ["counter", "count"],
+    "HobbyBoard": ["none", "none"],
+    "branch_0": ["presense", "present"],
+    "branch_1": ["presense", "present"],
+    "branch_2": ["presense", "present"],
+    "branch_3": ["presense", "present"],
+    "short_0": ["problem", "short"],
+    "short_1": ["problem", "short"],
+    "short_2": ["problem", "short"],
+    "short_3": ["problem", "short"],
 }
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_NAMES): {cv.string: cv.string},
         vol.Optional(CONF_MOUNT_DIR, default=DEFAULT_MOUNT_DIR): cv.string,
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=4304): cv.port,
     }
 )
 
 
+def hb_info_from_type(type="std"):
+    """Return the proper info array for the device type."""
+    if "std" in type:
+        return DEVICE_SENSORS
+    if "HobbyBoard" in type:
+        return HOBBYBOARD_EF
+
+
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the one wire Sensors."""
     base_dir = config.get(CONF_MOUNT_DIR)
+    owport = config.get(CONF_PORT)
+    owhost = config.get(CONF_HOST)
     devs = []
     device_names = {}
     if "names" in config:
         if isinstance(config["names"], dict):
             device_names = config["names"]
 
-    if base_dir == DEFAULT_MOUNT_DIR:
+    # We have an owserver on a remote(or local) host/port
+    if owhost:
+        try:
+            owproxy = protocol.proxy(host=owhost, port=owport)
+            devices = owproxy.dir()
+        except protocol.Error as exc:
+            _LOGGER.error(f"Cannot connect to owserver on {owhost}:{owport}, got:{exc}")
```

Comment:
Logging formatting should not use f-strings. We use the standard logging message formatting:
https://docs.python.org/3/library/logging.html#logging.Logger.debug

We want to pass the variables as parameters after the message string and use `%s` as formatting operator. This will defer the formatting until its needed.